framework/Metrics/SklMetric.py

```diff
@@ -83,53 +104,59 @@ def _localReadMoreXML(self,xmlNode):
-        if value.shape == (1,1):
-          return value[0]
-        else:
-          return value
-      else:
-        self.raiseAnError(IOError,'Metric SKL error: SKL metrics support only PointSets and not HistorySets')
-    else:
-      if self.metricType == 'mahalanobis':
-        covMAtrix = np.cov(x.T)
-        kwargs['VI'] = np.linalg.inv(covMAtrix)
+    #######################################################################################
+    # The inputs of regression metric, i.e. x, y should have shape (n_samples, n_outputs),
+    # and the outputs will have the shape (n_outputs).
+    # However, the inputs of paired metric, i.e. x, y should convert the shape to
+    # (n_outputs, n_samples), and the outputs will have the shape (n_outputs).
+    #######################################################################################
+    if isinstance(x,np.ndarray) and isinstance(y,np.ndarray):
+      assert(x.shape == y.shape, "Input data x, y should have the same shape")
+      if weights is not None and self.metricType[0] == 'regression' and 'sample_weight' not in self.distParams.keys():
+        self.distParams['sample_weight'] = weights
+      if self.metricType[0] == 'regression':
+        self.distParams['multioutput'] = 'raw_values'
       dictTemp = utils.mergeDictionaries(kwargs,self.distParams)
+      if self.metricType[0] == 'paired_distance':
+        if len(x.shape) == 1:
+          x = x.reshape(-1,1)
+          y = y.reshape(-1,1)
+        else:
+          # Transpose is needed, since paired_distance is operated on the 'row'
+          x = x.T
+          y = y.T
+      if axis == 1:
+        x = x.T
+        y = y.T
+        # check the dimension of weights
+        assert(x.shape[0] == len(weights), "'weights' should have the same length of the first dimension of input data")
+      elif axis != 0:
+        self.raiseAnError(IOError, "Valid axis value should be '0' or '1' for the evaluate method of metric", self. name, "value", axis, "is provided!")
       try:
-        if self.metricType in pairwise.kernel_metrics().keys():
-          value = pairwise.pairwise_kernels(X=x, metric=self.metricType, **dictTemp)
-        elif self.metricType in pairwise.distance_metrics().keys():
-          value = pairwise.pairwise_distances(X=x, metric=self.metricType, **dictTemp)
+        value = self.__class__.availMetrics[self.metricType[0]][self.metricType[1]](x, y, **dictTemp)
       except TypeError as e:
-        self.raiseAWarning('There are some unexpected keyword arguments found in Metric with type "', self.metricType, '"!')
+        self.raiseAWarning('There are some unexpected keyword arguments found in Metric with type "', self.metricType[1], '"!')
         self.raiseAnError(TypeError,'Input parameters error:\n', str(e), '\n')
-      if value.shape == (1,1):
-        return value[0]
-      else:
-        return value
+    else:
+      self.raiseAnError(IOError,'Input data type is not correct!')
```

Comment:
should it be an assert? It would be a developer error right?

Reply:
use assert now